src/Metrics/Reporter.h

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#pragma once
+
+#include "InternalMetric.h"
+#include "Sink.h"
+#include <asio.hpp>
+#include <map>
+#include <memory>
+#include <thread>
+
+namespace Metrics {
+
+class Reporter {
+public:
+  Reporter(std::unique_ptr<Sink> MetricSink, std::chrono::milliseconds Interval)
+      : MetricSink(std::move(MetricSink)), IO(), Period(Interval),
+        AsioTimer(IO, Period){};
+
```

Comment:
We should have a destructor that automatically stops the thread. It is too easy to forget to call `waitForStop()`.